components/App.jsx

```diff
@@ -0,0 +1,109 @@
+let React = require('react');
+let fx = require('money');
+
+let MainMenu = require('./MainMenu');
+let CurrencySelection = require('./CurrencySelection');
+let Balance = require('./Balance');
+let TransactionAction = require('./TransactionAction');
+let TransactionHistory = require('./TransactionHistory');
+
+fx.base = 'GBP';
+fx.rates = {
+  'BTC': 0.0073,
+  'EUR': 1.36,
+}
+
+// the App component contains all the other components
+let App = React.createClass({
+
+  getInitialState() {
+    let currency = localStorage.currency ? JSON.parse(localStorage.currency) : 'GBP';
+    let transactions = localStorage.transactions ? JSON.parse(localStorage.transactions) : [];
+    let balance = this.getNewBalance(transactions, currency);
+
+    return {
+      currency: currency,
+      transactions: transactions,
+      balance: balance,
+    };
+  },
+
+  // reset wallet to its initial state
+  handleReset() {
+    localStorage.clear();
+
+    this.setState(this.getInitialState());
+  },
+
+  // update currency state and recalculate the balance
+  handleCurrencyChange(currency) {
+    let balance = this.getNewBalance(this.state.transactions, currency);
+
+    this.setState({
+      currency: currency,
+      balance: balance,
+    });
+
+    localStorage.currency = JSON.stringify(currency);
+  },
+
+  // add and save new transaction
+  // recalculate balance
+  handleNewTransaction(transaction) {
+    transaction.currency = this.state.currency;
+
+    let transacs = this.state.transactions;
+    transacs.unshift(transaction);
+
+    let balance = this.getNewBalance(transacs, this.state.currency);
+
+    this.setState({
+      transactions: transacs,
+      balance: balance,
+    });
+
+    localStorage.transactions = JSON.stringify(transacs);
+  },
+
+  // return recalculated balance
+  getNewBalance(transactions, currency) {
+    let balance = 0;
+
+    transactions.forEach((t) => {
```

Comment:
I gave up on this one because I couldn't find a simple to accumulate the result when working with objects.